context/src/main/java/io/micronaut/runtime/converters/time/TimeConverterRegistrar.java

```diff
@@ -143,9 +143,9 @@ public void register(ConversionService<?> conversionService) {
             LocalDateTime.class,
             (object, targetType, context) -> {
                 try {
-                    DateTimeFormatter formatter = resolveFormatter(context);
-                    LocalDateTime result = LocalDateTime.parse(object, formatter);
-                    return Optional.of(result);
+                    return Optional.of(resolveFormatter(context)
+                            .map(formatter -> LocalDateTime.parse(object, formatter))
+                            .orElse(LocalDateTime.parse(object)));
```

Comment:
I believe `.orElseGet(() -> LocalDateTime.parse(object))` would be more appropriate.